Libraries/pbxbuild/Sources/Tool/CopyResolver.cpp

```diff
@@ -77,13 +78,20 @@ ResolveInternal(
     Tool::OptionsResult options = Tool::OptionsResult::Create(toolEnvironment, toolContext->workingDirectory(), nullptr);
     Tool::Tokens::ToolExpansions tokens = Tool::Tokens::ExpandTool(toolEnvironment, options, std::string(), args);
 
+    std::vector<std::string> toolInputPaths = toolEnvironment.inputs(toolContext->workingDirectory());
+
     // TODO(grp): This should be generic for all tools.
     std::vector<Tool::Invocation::DependencyInfo> dependencyInfo;
     if (tool->deeplyStatInputDirectories()) {
         for (std::string const &inputPath : inputPaths) {
             /* Create a dependency info file to track the input directory contents. */
-            auto info = Tool::Invocation::DependencyInfo(dependency::DependencyInfoFormat::Directory, inputPath);
-            dependencyInfo.push_back(info);
+            libutil::DefaultFilesystem filesystem = libutil::DefaultFilesystem();
```

Comment:
This is bad, it makes this tool not self-contained. If this is needed here, use `Filesystem::GetDefaultUNSAFE()` to indicate it needs to be replaced by passing down a filesystem.

On the other hand, I think it'd be better to change the dependency info tool to ignore files rather than error. Then this code doesn't have to read the filesystem at all. (I'd prefer if filesystem reads are only done during execution, otherwise Ninja generation could be wrong if this isn't added to the regeneration rule.)